src/util/simplify_expr.cpp

```diff
@@ -2047,6 +2047,113 @@ simplify_exprt::simplify_complex(const unary_exprt &expr)
   return unchanged(expr);
 }
 
+simplify_exprt::resultt<>
+simplify_exprt::simplify_overflow_binary(const binary_exprt &expr)
+{
+  // zero is a neutral element for all operations supported here
+  if(expr.op0().is_zero() || expr.op1().is_zero())
+    return false_exprt{};
```

Comment:
What about the following? This appears to have a zero and also an overflow.
unsigned int x = 0;
unsigned int y = 2;
x = x - y;

Reply:
Good catch!

Reply:
Thanks, and added as a test!